contracts/beacon/Voucher.sol

```diff
@@ -164,11 +184,69 @@ contract Voucher is OwnableUpgradeable, IVoucher {
             workerpoolOrder,
             requestOrder
         );
-        $._sponsoredAmounts[dealId] = sponsoredAmount;
+        $._voucherMatchedDeals[dealId].exists = true;
+        $._voucherMatchedDeals[dealId].sponsoredAmount = uint64(sponsoredAmount);
         emit OrdersBoostMatchedWithVoucher(dealId);
         return dealId;
     }
 
+    function claim(bytes32 dealId, uint256 index) external {
+        VoucherStorage storage $ = _getVoucherStorage();
+        IVoucherHub voucherHub = IVoucherHub($._voucherHub);
+        address iexecPoco = voucherHub.getIexecPoco();
+        // Check if task is already claimed.
+        bytes32 taskId = keccak256(abi.encodePacked(dealId, index));
+        require(!$._claimedTasks[taskId], "Voucher: task already claimed");
+        // Get deal details from PoCo.
+        (bool isBoostDeal, address requester, uint256 volume, uint256 dealPrice) = _getDealDetails(
+            iexecPoco,
+            dealId
+        );
+        require(requester != address(0), "Voucher: deal not found");
```

Comment:
```suggestion
        require(requester != address(0), "Voucher: deal not founded");
```

Reply:
I think "found" is the correct form here.